### ISSUE TITLE
[ERROR] : Error: Unknown error

### ISSUE THREAD

**Issue by shir** [kwent/syno#25]
Hi. Thanks for tool. But have some Unknown error when trying to get list of tasks from download station:

```
$ ./node_modules/.bin/syno --url http://USER:PASSWORD@192.168.0.37:5000 --debug dl getTaskList
[DEBUG] : Params URL detected : http://USER:PASSWORD@192.168.0.37:5000.
[DEBUG] : DSM Connection URL configured : http://USER:PASSWORD@192.168.0.37:5000
[DEBUG] : Account: USER
[DEBUG] : Password: PASSWORD
[DEBUG] : Host: 192.168.0.37
[DEBUG] : Port: 5000
[DEBUG] : API: 6.0
[DEBUG] : Ignore certificate errors: false
[DEBUG] : DSM Download Station API command selected
[DEBUG] : Method name configured : getTaskList
REQUEST { rejectUnauthorized: true,
  jar: true,
  json: true,
  url: 'http://192.168.0.37:5000/webapi/auth.cgi',
  qs: 
   { api: 'SYNO.API.Auth',
     version: 3,
     method: 'login',
     account: 'USER',
     passwd: 'PASSWORD',
     session: 'SYNO_SESSION_1478032259136' },
  callback: [Function] }
REQUEST make request http://192.168.0.37:5000/webapi/auth.cgi?api=SYNO.API.Auth&version=3&method=login&account= USER&passwd= PASSWORD&session=SYNO_SESSION_1478032259136
REQUEST onRequestResponse http://192.168.0.37:5000/webapi/auth.cgi?api=SYNO.API.Auth&version=3&method=login&account= USER&passwd= PASSWORD&session=SYNO_SESSION_1478032259136 200 { server: 'nginx',
  date: 'Tue, 01 Nov 2016 20:20:37 GMT',
  'content-type': 'text/plain; charset="UTF-8"',
  'transfer-encoding': 'chunked',
  connection: 'close',
  vary: 'Accept-Encoding',
  'x-content-type-options': 'nosniff',
  'x-xss-protection': '1; mode=block',
  p3p: 'CP="IDC DSP COR ADM DEVi TAIi PSA PSD IVAi IVDi CONi HIS OUR IND CNT"',
  'set-cookie': [ 'id=wO12pG8X3GfQQ1470M4N227000;expires=Tue, 08-Nov-2016 20:20:37 GMT;path=/' ] }
REQUEST reading response's body
REQUEST finish init function http://192.168.0.37:5000/webapi/auth.cgi?api=SYNO.API.Auth&version=3&method=login&account= USER&passwd= PASSWORD&session=SYNO_SESSION_1478032259136
REQUEST response end http://192.168.0.37:5000/webapi/auth.cgi?api=SYNO.API.Auth&version=3&method=login&account= USER&passwd= PASSWORD&session=SYNO_SESSION_1478032259136 200 { server: 'nginx',
  date: 'Tue, 01 Nov 2016 20:20:37 GMT',
  'content-type': 'text/plain; charset="UTF-8"',
  'transfer-encoding': 'chunked',
  connection: 'close',
  vary: 'Accept-Encoding',
  'x-content-type-options': 'nosniff',
  'x-xss-protection': '1; mode=block',
  p3p: 'CP="IDC DSP COR ADM DEVi TAIi PSA PSD IVAi IVDi CONi HIS OUR IND CNT"',
  'set-cookie': [ 'id=wO12pG8X3GfQQ1470M4N227000;expires=Tue, 08-Nov-2016 20:20:37 GMT;path=/' ] }
REQUEST end event http://192.168.0.37:5000/webapi/auth.cgi?api=SYNO.API.Auth&version=3&method=login&account= USER&passwd= PASSWORD&session=SYNO_SESSION_1478032259136
REQUEST has body http://192.168.0.37:5000/webapi/auth.cgi?api=SYNO.API.Auth&version=3&method=login&account= USER&passwd= PASSWORD&session=SYNO_SESSION_1478032259136 61
REQUEST emitting complete http://192.168.0.37:5000/webapi/auth.cgi?api=SYNO.API.Auth&version=3&method=login&account= USER&passwd= PASSWORD&session=SYNO_SESSION_1478032259136
REQUEST { rejectUnauthorized: true,
  jar: true,
  json: true,
  url: 'http://192.168.0.37:5000/webapi/entry.cgi',
  qs: 
   { api: 'SYNO.DownloadStation2.Task.List',
     version: '1',
     method: 'get' },
  callback: [Function] }
REQUEST make request http://192.168.0.37:5000/webapi/entry.cgi?api=SYNO.DownloadStation2.Task.List&version=1&method=get
REQUEST onRequestResponse http://192.168.0.37:5000/webapi/entry.cgi?api=SYNO.DownloadStation2.Task.List&version=1&method=get 200 { server: 'nginx',
  date: 'Tue, 01 Nov 2016 20:20:38 GMT',
  'content-type': 'application/json; charset="UTF-8"',
  'transfer-encoding': 'chunked',
  connection: 'close',
  vary: 'Accept-Encoding',
  'x-content-type-options': 'nosniff',
  'x-xss-protection': '1; mode=block',
  'cache-control': 'max-age=0, no-cache, no-store, must-revalidate',
  pragma: 'no-cache',
  expires: '0' }
REQUEST reading response's body
REQUEST finish init function http://192.168.0.37:5000/webapi/entry.cgi?api=SYNO.DownloadStation2.Task.List&version=1&method=get
REQUEST response end http://192.168.0.37:5000/webapi/entry.cgi?api=SYNO.DownloadStation2.Task.List&version=1&method=get 200 { server: 'nginx',
  date: 'Tue, 01 Nov 2016 20:20:38 GMT',
  'content-type': 'application/json; charset="UTF-8"',
  'transfer-encoding': 'chunked',
  connection: 'close',
  vary: 'Accept-Encoding',
  'x-content-type-options': 'nosniff',
  'x-xss-protection': '1; mode=block',
  'cache-control': 'max-age=0, no-cache, no-store, must-revalidate',
  pragma: 'no-cache',
  expires: '0' }
REQUEST end event http://192.168.0.37:5000/webapi/entry.cgi?api=SYNO.DownloadStation2.Task.List&version=1&method=get
REQUEST has body http://192.168.0.37:5000/webapi/entry.cgi?api=SYNO.DownloadStation2.Task.List&version=1&method=get 86
REQUEST emitting complete http://192.168.0.37:5000/webapi/entry.cgi?api=SYNO.DownloadStation2.Task.List&version=1&method=get
[ERROR] : Error: Unknown error
REQUEST { rejectUnauthorized: true,
  jar: true,
  json: true,
  url: 'http://192.168.0.37:5000/webapi/auth.cgi',
  qs: 
   { api: 'SYNO.API.Auth',
     version: 3,
     method: 'logout',
     session: 'SYNO_SESSION_1478032259136' },
  callback: [Function] }
```

I have same error if try to make `listFiles`. The only method I get to work is `getInfo`.

Just have an idea: could it be related to node.js 7?

**Comment by kwent** [comment]
What version of DSM and DownloadStation do you have installed ?

**Comment by shir** [comment]
@kwent 

DSM 6.0.2-8451 Update 2, Download Station: 3.7.3-3409. Latest available.

**Comment by kwent** [comment]
Hi @shir, the API compiled in this package for Download Station is `3.7.1-3282`. Maybe that would do it if they considerably change their API. I would have to compile this package against your version package to make it work. PR is welcome !

**Comment by shir** [comment]
@kwent do you have any guide how I may compile for new API version? I've tried to figure it out by myself but without success.

**Comment by shir** [comment]
@kwent I made some investigation using tcpdump and found response to my request (why don't you show response in debug mode? It may be very helpful):

``` json
{"error":{"code":120,"errors":{"name":"list_id","reason":"required"}},"success":false}
```

So I figured out that I use wrong method. And method that I need is `listTasks`. But then I try to use this method I see request to url `http://192.168.0.35:5000/webapi/DownloadStation/xunlei/dl_queue.cgi?api=SYNO.DownloadStation.Xunlei.Task&version=1&method=list` and it returns error 403.

Why `Xunlei`? I don't even enable this service in Download Station (that's why I have error 403). I expect what `listTasks` will make request to url `http://192.168.0.35:5000/webapi/DownloadStation/task.cgi?api=SYNO.DownloadStation.Task&version=1&method=list`. Is it a bug? Or should I use some other method to get list of tasks?
